internal/queryexecutor/range_query_planner.go

```diff
@@ -0,0 +1,117 @@
+package queryexecutor
+
+import (
+	"errors"
+
+	"github.com/IBM-Blockchain/bcdb-server/internal/stateindex"
+	"github.com/IBM-Blockchain/bcdb-server/pkg/constants"
+	"github.com/IBM-Blockchain/bcdb-server/pkg/types"
+)
+
+type rangeQueryPlan struct {
+	startKey *stateindex.IndexEntry
+	endKey   *stateindex.IndexEntry
+}
+
+func createQueryPlan(attribute string, conds *attrConditions) (*rangeQueryPlan, error) {
+	// we assume this function to get only valid conditions
+	//   - eq and no other conditions
+	//   - lt and lte do not appear together
+	//   - gt and gte do not appear together
+
+	if conds.valueType == types.Type_NUMBER {
+		// TODO: support number values
+		return nil, errors.New("numbers are not supported currently")
+	}
+
+	for c := range conds.conditions {
+		if c == constants.NotEqual {
+			// TODO: support not equal
+			return nil, errors.New(constants.NotEqual + " is not supported currently")
+		}
+	}
+
+	p := &rangeQueryPlan{
+		startKey: &stateindex.IndexEntry{
+			Attribute: attribute,
+			Type:      conds.valueType,
+		},
+		endKey: &stateindex.IndexEntry{
+			Attribute: attribute,
+			Type:      conds.valueType,
+		},
+	}
+
+	if len(conds.conditions) == 1 {
+		for c, v := range conds.conditions {
+			setPlanForSingleCondition(c, v, p)
+			return p, nil
+		}
```

Comment:
If `len==1`, isn't it clearer to just access `conds.conditions[0]` ?

Reply:
condition is a map, that's why.